modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -639,7 +768,21 @@ public void destroy() {
     /** {@inheritDoc} */
     @Override
     public void close() throws Exception {
-        IgniteUtils.closeAll(writeOpts, upperBound);
+        for (CompletableFuture<Void> future : flushFutures.values()) {
+            future.completeExceptionally(new StorageException("Can't complete flush operation, partition is being stopped."));
```

Comment:
Should we cancel the future instead?

Reply:
`java.util.concurrent.CompletionStage#whenComplete` is used on these future later. I'm not sure how it'll work with cancelling. Can you assist?

Reply:
It will be completed exceptionally with a `CancellationException`

Reply:
Ok, that's cool